extensions/wikia/AdEngine/js/slot/floatingMedrec.js

```diff
@@ -1,85 +1,133 @@
+			refreshInfo = {
+				refreshAdPos: 0,
+				lastRefreshTime: new Date(),
+				refreshNumber: 0,
+				adVisible: true,
+				heightScrolledThreshold: 10,
+				refreshDelay: 10000
+			},
 			slotName = 'INCONTENT_BOXAD_1',
-			startPosition,
-			stopPoint,
-			stopPosition,
-			$adSlot = $('<div class="wikia-ad"></div>').attr('id', slotName),
-			$footer = $('#WikiaFooter'),
-			$placeHolder = $('#WikiaAdInContentPlaceHolder'),
-			$win = $(win);
-
-		function getStartPosition(placeHolder) {
-			return parseInt(placeHolder.offset().top, 10) - globalNavigationHeight - margin;
+			startPosition = placeHolderElement.offsetTop,
+			swapRecirculationAndAd;
+
+		adSlot.className = 'wikia-ad';
+		adSlot.setAttribute('id', slotName);
+
+		function hasUserScrolledEnoughDistance(currentHeightPosition) {
+			var heightScrolled = Math.abs(currentHeightPosition - refreshInfo.refreshAdPos);
+			return heightScrolled > refreshInfo.heightScrolledThreshold;
 		}
 
-		function getStopPosition(ad, footer, leftSkyscraper3) {
-			stopPoint = parseInt(footer.offset().top, 10);
+		function wasItEnoughTimeSinceLastRefresh() {
+			var timeDifference = (new Date()) - refreshInfo.lastRefreshTime;
+			return timeDifference > refreshInfo.refreshDelay;
+		}
 
-			if (leftSkyscraper3.length && parseInt(leftSkyscraper3.height(), 10) !== 0) {
-				stopPoint = parseInt(leftSkyscraper3.offset().top, 10);
-			}
+		function isMaxNumberOfRefreshReached() {
+			return refreshInfo.refreshNumber >= maxChanges;
+		}
 
-			return stopPoint - globalNavigationHeight - 2 * margin - ad.height();
+		function isUAPFloatingMedrecVisible() {
+			var isAdVisible = refreshInfo.adVisible && refreshInfo.refreshNumber !== 0;
+			return uapContext.isUapLoaded() && isAdVisible;
 		}
 
-		handleFloatingMedrec = throttle(function () {
-			var scrollTop = $win.scrollTop();
+		function shouldSwitchModules(currentHeightPosition) {
+			return hasUserScrolledEnoughDistance(currentHeightPosition) &&
+				wasItEnoughTimeSinceLastRefresh() &&
+				!isMaxNumberOfRefreshReached() &&
+				!isUAPFloatingMedrecVisible();
+		}
 
-			startPosition = getStartPosition($placeHolder);
-			stopPosition = getStopPosition($adSlot, $footer, $(leftSkyscraper3Selector));
-			isEnoughSpace = stopPosition - startPosition > minDistance;
+		function updateModulesRefreshInformation(currentHeightPosition) {
+			refreshInfo.lastRefreshTime = new Date();
+			refreshInfo.refreshAdPos = currentHeightPosition;
+			refreshInfo.refreshNumber++;
+		}
 
-			if (enabled && !isEnoughSpace) {
-				log(['handleFloatingMedrec',
-					'Disabling floating medrec: not enough space in right rail'], 'debug', logGroup);
+		function showRecirculation() {
+			recirculationElement.style.display = 'block';
+		}
 
+		function hideRecirculation() {
+			recirculationElement.style.display = 'none';
+		}
 
-				$adSlot.css({
-					visibility: 'hidden'
+		function refreshAd() {
+			return new Promise(function (resolve, reject) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "reject". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

[project-id]: wikia:app